packages/core-sdk/resolvers/token-resolvers/resolvers/1inch/src/query/__tests__/e2e/apiCalls.ts

```diff
@@ -0,0 +1,74 @@
+import { InvokeApiResult, Web3ApiClient } from "@web3api/client-js";
+
+import { Interface_TokenComponent } from "../../w3";
+
+export const isValidProtocolToken = async (
+  tokenAddress: string,
+  protocolId: string,
+  protocolEnsUri: string,
+  client: Web3ApiClient,
+): Promise<InvokeApiResult<boolean>> => {
+  return client.invoke<boolean>({
+    uri: protocolEnsUri,
+    module: "query",
+    method: "isValidProtocolToken",
+    input: {
+      tokenAddress,
+      protocolId,
+    },
+    config: {
+      envs: [
+        {
+          uri: protocolEnsUri,
+          query: {
+            connection: {
+              networkNameOrChainId: "MAINNET",
+            },
+          },
+        },
+      ],
+    },
+  });
+};
+
+export const getTokenComponents = async (
+  tokenAddress: string,
+  tokenEnsUri: string,
+  protocolEnsUri: string,
+  client: Web3ApiClient,
+): Promise<InvokeApiResult<Interface_TokenComponent>> => {
+  return client.invoke<Interface_TokenComponent>({
+    uri: protocolEnsUri,
+    module: "query",
+    method: "getTokenComponents",
+    input: {
+      tokenAddress: tokenAddress,
+    },
+    config: {
+      redirects: [
+        {
+          from: "ens/token.defiwrapper.eth",
+          to: tokenEnsUri,
+        },
+      ],
+      envs: [
+        {
+          uri: protocolEnsUri,
+          query: {
+            connection: {
+              networkNameOrChainId: "MAINNET",
+            },
+          },
+        },
+        {
+          uri: "ens/token.defiwrapper.eth",
+          query: {
+            connection: {
+              networkNameOrChainId: "MAINNET",
```

Comment:
With the current setup, we need to set the `QueryEnv.connection` property on each wrapper being used, and not just the "root" wrapper (i.e. root "defiwrapper" aggregation wrapper).

How can we get around this? I suggest the following architecture:
Only set the Ethereum wrapper's `QueryEnv.connection` property, all calls to the DeFi Wrapper will use what's already been set. In order to support multiple async network queries for token balances, we can use the "query time environment configuration" feature that's been added to the client.

This to me makes the most sense because it is good separation of responsibilities. If the DeFiWrapper needs to know what network is begin used, it can simply query the ethereum wrapper to find out. It also makes the most sense from a user developer experience perspective, where app devs only have to worry about configuring ethereum, and can query other wrappers that use ethereum without needed to configure them as well.

Reply:
Additionally it might make sense to set `QueryEnv.connection` on the root DeFi Wrapper's env.

Reply:
I don't think we need it at root level, since we already have support for getting network info from ethereum plugin:
`Ethereum_Query.getNetwork`

Reply:
That sounds great. A much cleaner approach. Thanks!